library/oneview_server_profile.py

```diff
@@ -16,236 +16,260 @@
 ###
 
-import hpOneView as hpov
-from hpOneView.exceptions import *
 from ansible.module_utils.basic import *
-
-__author__ = 'ChakruHP'
+from hpOneView.oneview_client import OneViewClient
 
 DOCUMENTATION = '''
 ---
 module: oneview_server_profile
-short_description: Selects a server hardware automatically based on the server hardware template.
+short_description: Manage OneView Server Profile resources.
 description:
-    - Manage the servers lifecycle with OneView Server Profiles using an existing server profile template.
+    - Manage the servers lifecycle with OneView Server Profiles using an existing server profile template. On 'present'
+      state it selects a server hardware automatically based on the server profile template.
 requirements:
     - "python >= 2.7.9"
     - "hpOneView"
+author: "Chakravarthy Racharla"
 options:
-  oneview_host:
-    description:
-      - OneView appliance IP or hostname.
-    required: true
-  username:
+  config:
     description:
-      - Username that will be used to authenticate on the provided OneView appliance.
+      - Path to a .json configuration file containing the OneView client configuration.
     required: true
-  password:
+  state:
     description:
-      - Password that will be used to authenticate on the provided OneView appliance.
-    required: true
+      - Indicates the desired state for the Server Profile resource by the end of the playbook execution.
+        'present' will ensure data properties are compliant to OneView.
+        'absent' will remove the resource from OneView, if it exists.
+        'powered_off' requests a power operation to change the power state of the physical server to Off.
+        'powered_on' requests a power operation to change the power state of the physical server to On.
+        'no-op' gather facts about the Server Profile
+    default: present
+    choices: ['present', 'powered_off', 'absent', 'powered_on', 'no-op']
   server_template:
     description:
       - Name of the server profile template that will be used to provision the server profiles.
-    required: true
+    required: false
   name:
     description:
       - Name of the server profile that will be created or updated.
     required : true
-  state:
-    description:
-      - Desired state for the server profile by the end of the playbook execution.
-    default: present
-    choices: ['present', 'powered_off', 'absent', 'powered_on', 'restarted']
+notes:
+    - "A sample configuration file for the config parameter can be found at:
+       https://github.com/HewlettPackard/oneview-ansible/blob/master/examples/oneview_config-rename.json"
 '''
 
 EXAMPLES = '''
-- oneview_server_profile:
-    oneview_host: <ip>
-    username: oneview_username
-    password: oneview_password
+- name: Create a Server Profile from a Server Profile Template
+  oneview_server_profile:
+    config: "{{ config }}"
     server_template: Compute-node-template
     name: <server-profile-name>
```

Comment:
Some questions about the examples:
1. Why the playbook https://github.com/GustavoHennig/oneview-ansible/blob/server_profile/examples/oneview_server_profile.yml was not updated yet? Looks like it's outdated,  since it's still authenticating in the previous style, using a username and password instead of a json file.
2. Why this documentation has only one example?